homeassistant/helpers/service.py

```diff
@@ -841,3 +845,39 @@ async def execute_service(self, service_call: ServiceCall) -> None:
             async with self._service_condition:
                 self._service_running = False
                 self._service_condition.notify_all()
+
+
+@bind_hass
+@callback
+def async_get_config_entry_from_call(
+    hass: HomeAssistant, call: ServiceCall
+) -> ConfigEntry | None:
+    """Get the config entry related to a particular service call.
```

Comment:
I think, honestly, that this will induce a pattern we should not promote.

IMHO, this should use/need a config entry selector in the future. I feel like these are all workarounds for that.

Reply:
Fair!